### ISSUE TITLE
assert (boxes1[:, 2:] >= boxes1[:, :2]).all() 

### ISSUE THREAD

**Issue by Puranjay-del-Mishra** [facebookresearch/detr#509]
I get the above error in the boxs_op.py file, when i set the default batch size to 1 from 2  in line 63. All the coordinates of boxes1 become NaN for some reason. 
This is not the only way to get this error though.I found these bugs when I was trying to reduce the size of input due to the GPU limitations. I also tried to reduce the max_size and the scale list to a size of 64. This too gave me the NaN error after one epoch. I hope this is addressed soon.
 Thank you in advance !
@fmassa , @zhiqwang , @lessw2020 

Update 1 (2-06-2022)
I tried to debug the pipeline by printing out the intermediate values in DETR. I found that on the second training step, the feature extractor backbone(using the default resnet50) gave the output as NaN with the input being a tensor with float values and appropriate shape. This can be done by print the output features obtained in line 123 in the deformable_detr.py file . 
Update 2 (03-06-2022)
I debugged it a bit further and found that the BackboneBase class's forwared method's output becomes NaN after training of the first batch. printing out the values of tensor_list argument and then printing out before returning it confirms this.

**Comment by Puranjay-del-Mishra** [comment]
I would really appreciate if this was addressed soon, it is a big bottleneck in something I want to try out.

**Comment by Puranjay-del-Mishra** [comment]
No response yet :(

**Comment by Puranjay-del-Mishra** [comment]
There's no activity from the meta team in the issue section of DETR. Unfortunate

**Comment by EMU1337X** [comment]
Hi bro, I meet the same prob when I use **deformable** DETR to train my custom dataset, I'm pretty sure my data processing is absolutely ok and fit detr model input. 

When this error occurs the output box of model is NaN, but **before that iter I saved the model which performs just ok on testset**, that means the model did learn something useful from the data, very strange.

My environment: 2 x GTX1080Ti, torch1.7, cuda101, using DistributedDataParallel.

training set have about 10^5 samples and assembled into 10^4 batches with batch_size at 10. **When training process proceeds to about 450 batch it assert that error**, very confused.

**Comment by EMU1337X** [comment]
@fmassa , sir could u plz give us some advice?

**Comment by EMU1337X** [comment]
@fmassa @lessw2020 , sir do u have any idea on this?

**Comment by Puranjay-del-Mishra** [comment]
@EMU1337X This happens with me on the second epoch of the COCO dataset. Your feedback makes things even more interesting.

**Comment by alcinos** [comment]
Closing as duplicate of #101. See the solutions proposed in that issue. The most likely reason is that your training is diverging. Try reducing the learning rate. Also make sure your batch size is big enough, less than 8 is unlikely to work.

**Comment by Puranjay-del-Mishra** [comment]
> Closing as duplicate of #101. See the solutions proposed in that issue. The most likely reason is that your training is diverging. Try reducing the learning rate. Also make sure your batch size is big enough, less than 8 is unlikely to work.

I did try out the solution and as the comments suggest, reduction of the learning rate just leads to this issue popping out later in the training loop, reducing the LR significantly did nothing for me. The number of classes and other hyperparameters were left undisturbed, the only thing that changed was the batch size(changed it from 2 to 1).
